src/Stratis.Bitcoin.Features.Consensus/Rules/CommonRules/PosCoinviewRule.cs

```diff
@@ -4,133 +4,93 @@
-    /// <item>Coinbase transaction's (first transaction in the block with no inputs) timestamp MUST be inside interval ['BlockTime' - 15; 'BlockTime'].</item>
-    /// <item>Coinstake transaction's (second transaction in the block with at least one input and at least 2 outputs and first output being empty) timestamp
-    /// MUST be equal to 'BlockTime' and it MUST have lower 4 bits set to 0 (i.e. be divisible by 16) - see <see cref="StakeTimestampMask"/>.</item>
-    /// <item>Block's header 'nBits' field MUST be set to the correct POS target value.</item>
-    /// <item>All transactions in the block must be final, which means their 'nLockTime' is either zero, or it is lower than current block's height
-    /// or node's 'AdjustedTime'. 'AdjustedTime' is the synchronized time among the node and its peers.</item>
-    /// <item>Coinstake transaction MUST be signed correctly.</item>
-    /// <item>Coinstake transaction's kernel (first) input MUST not be created within last <see cref="PosConsensusOptions.StakeMinConfirmations"/> blocks,
-    /// i.e. it MUST have that many confirmation at least.</item>
-    /// <item>Coinstake transaction's kernel must meet the staking target using this formula:
-    /// <code>hash(stakeModifierV2 + stakingCoins.Time + prevout.Hash + prevout.N + transactionTime) &lt; target * weight</code>
-    /// <para>
-    /// where 'stakingCoins' is the coinstake's kernel UTXO, 'prevout' is the kernel's output in that transaction,
-    /// 'prevout.Hash' is the hash of that transaction; 'transactionTime' is coinstake's transaction time; 'target' is the target as
-    /// in 'Bits' block header; 'weight' is the value of the kernel's input.
-    /// </para>
-    /// </item>
-    /// <item>Block's height MUST NOT be more than 500 blocks back - i.e. reorganizations longer than 500 are not allowed.</item>
-    /// <item>Coinbase 'scriptSig' starts with serialized block height value. This means that coinbase transaction commits to the height of the block it appears in.</item>
-    /// </list>
-    /// </remarks>
-    public class PosConsensusValidator : PowConsensusValidator, IPosConsensusValidator
+    [ExecutionRule]
+    public class PosCoinviewRule : PowCoinviewRule
     {
         /// <summary>PoS block's timestamp mask.</summary>
         /// <remarks>Used to decrease granularity of timestamp. Supposed to be 2^n-1.</remarks>
         public const uint StakeTimestampMask = 0x0000000F;
 
-        /// <summary>Drifting Bug Fix, hardfork on Sat, 19 Nov 2016 00:00:00 GMT.</summary>
-        public const long DriftingBugFixTimestamp = 1479513600;
-
-        /// <summary>Instance logger.</summary>
-        private readonly ILogger logger;
-
         /// <summary>Provides functionality for checking validity of PoS blocks.</summary>
-        public IStakeValidator StakeValidator { get; }
+        public IStakeValidator StakeValidator { get; set; }
```

Comment:
StakeValidator should be a private member

Reply:
agree

Reply:
private member, not private setter

Reply:
yep this is a private member now